tests/migrations/test_loader.py

```diff
@@ -464,3 +465,15 @@ def test_loading_squashed_ref_squashed(self):
             ('app1', '4_auto'),
         }
         self.assertEqual(plan, expected_plan)
+
+    def test_readonly_database(self):
+        """
+        MigrationLoader.check_consistent_history() shouldn't choke on
+        read-only databases (possibly without a django_migrations table).
+        """
+        with mock.patch.object(
+                MigrationRecorder, 'ensure_schema',
+                autospec=True,
```

Comment:
The autospec here cause the __call__ on the mock object to have the same signature as ensure_schema. This we necessary so that the migration recorder object (self) ends up being passed to patched_ensure_schema.